joblib/test/test_memory.py

```diff
@@ -162,14 +162,12 @@ def name_collision(x):
 
     b = name_collision
 
-    with warnings.catch_warnings(record=True) as w:
-        # Cause all warnings to always be triggered.
-        warnings.simplefilter("always")
+    with warns(JobLibCollisionWarning) as warninfo:
```

Comment:
I changed completely my mind on this one, `warninfo` is a nice parallel to `excinfo` and is more specific than `record`. So I pushed some changes to revert back to the original state, sorry about that.